lib/Backend/GlobOpt.h

```diff
@@ -318,18 +323,14 @@ template<> ValueNumber JsUtil::ValueToKey<ValueNumber, Value *>::ToKey(Value *co
 
 class IntConstantValueInfo : public ValueInfo
 {
-private:
-    const int32 intValue;
 
-protected:
-    IntConstantValueInfo(const int32 intValue)
-        : ValueInfo(GetInt(IsTaggable(intValue)), ValueStructureKind::IntConstant),
-        intValue(intValue)
+public:
+    static IntConstantValueInfo *New(JitArenaAllocator *const allocator, const int32 intValue)
```

Comment:
I think Louis is right and we should use your template version of this. Otherwise we risk introducing memory regression for js